nameko_sqlalchemy/session_scope.py

```diff
@@ -0,0 +1,48 @@
+from contextlib import contextmanager
+
+from nameko.extensions import DependencyProvider
+from sqlalchemy import create_engine
+from sqlalchemy.orm import sessionmaker
+
+
+DB_URIS_KEY = 'DB_URIS'
+
+
+class DatabaseSessionScope(DependencyProvider):
+
+    def __init__(self, declarative_base):
+        self.declarative_base = declarative_base
+
+    def setup(self):
+        service_name = self.container.service_name
+        decl_base_name = self.declarative_base.__name__
+
+        uri_key = '{}:{}'.format(service_name, decl_base_name)
+
+        db_uris = self.container.config[DB_URIS_KEY]
+        self.db_uri = db_uris[uri_key].format({
+            'service_name': service_name,
+            'declarative_base_name': decl_base_name,
+        })
+        self.engine = create_engine(self.db_uri)
+        self.Session = sessionmaker(bind=self.engine)
```

Comment:
Is this any different from the existing `DatabaseSession.setup()`? If not, why not make this a subclass?

Reply:
Good point, I started separately, but it makes sense to reuse as much as possible

I put both in the same module